app/src/main/cpp/skyline/input/touch.cpp

```diff
@@ -26,17 +26,23 @@ namespace skyline::input {
         entry.localTimestamp = lastEntry.localTimestamp + 1;
         entry.touchCount = points.size();
 
-        for (size_t i{}; i < points.size(); i++) {
+        for (size_t i{}; i < points.size() && i < MaxTouchPoints; i++) {
             const auto &host{points[i]};
             auto &guest{entry.data[i]};
-            guest.index = static_cast<u32>(i);
+            guest.attribute.raw = static_cast<u32>(host.attribute);
+            guest.index = static_cast<u32>(host.id);
             guest.positionX = static_cast<u32>(host.x);
             guest.positionY = static_cast<u32>(host.y);
             guest.minorAxis = static_cast<u32>(host.minor);
             guest.majorAxis = static_cast<u32>(host.major);
             guest.angle = host.angle;
         }
 
+        // Clear unused touch points
+        for (size_t i{points.size()}; i < MaxTouchPoints; i++) {
```

Comment:
@bylaws Any idea if this can be replaced by magic C++ ranges stuff?

Reply:
Not really since it only covers a subspan and a solution like `ranges::generate(span(data).subspan(points.size()), []{return TouchPoint{};})` at least to me seems uglier

Reply:
Ah well, I agree.